ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoicePayloadExecutor.java

```diff
@@ -79,6 +79,10 @@ public boolean optimisticallyExecute(
                       if (result.hasValidStatus()) {
                         return transitionBlockValidator.verifyTransitionBlock(
                             latestExecutionPayloadHeader, block);
+                      } else if (result.hasInvalidStatus()
+                          && result.getLatestValidHash().isPresent()) {
+                        return SafeFuture.completedFuture(
+                            new PayloadValidationResult(block.getParentRoot(), result));
```

Comment:
I'm not sure this is correct. You are assuming that the parent of the current block is invalid, by treating it as an `invalidTransitionBlock`. I think we need to manage it differently. By handling it in `ForkChoice#importBlockAndState` and passing the current valid optimistic head and `lastValidBlockHash` to a new method in `ProtoArray`  that searches the payload in protoarray similarly to what is already done in `markNodeInvalid`.

Reply:
You are right, I'm marking parent as invalid which could be not true, I will re-implement it in proposed way